docs/architecture/overview.md

```diff
@@ -0,0 +1,115 @@
+# Pulumi Terraform Bridge Architecture Overview
+
+This document captures the high–level mental model maintainers use to reason about the Pulumi Terraform Bridge. It is
+intended as a launchpad: each section links to the code, docs, or playbooks that provide deeper detail.
+
+## System Goals
+
+The bridge has two equally important responsibilities:
+
+1. **Build-time generation** – Translate a Terraform provider's schema and metadata into a Pulumi schema, SDKs, and docs so Pulumi users can program against the provider.
+2. **Runtime translation** – Act as a Pulumi resource provider that speaks Pulumi's gRPC protocol while driving the
+   Terraform provider implementation under the hood.
+
+A change to either half can impact the other. Treat the bridge as one product with two tightly-coupled pipelines.
+
+## Major Components
+
+| Layer | Location | Role |
+| ----- | -------- | ---- |
+| Build-time | `pkg/tfgen`, `docs/guides/*`, `pkg/convert`, `pkg/tf2pulumi` | Introspect Terraform provider schemas, emit Pulumi schema, SDKs, and docs. |
+| Runtime (SDKv2) | `pkg/tfbridge`, `pkg/tfshim/sdk-v{1,2}`, `pkg/providerserver` | Drive Terraform Plugin SDK providers via Pulumi RPC entry points. |
+| Runtime (PF) | `pkg/pf/*`, `pkg/tfshim/schema` | Bridge Terraform Plugin Framework providers. |
+| Hybrid / muxing | `pkg/x/muxer`, `docs/guides/upgrade-sdk-to-mux.md` | Compose multiple runtime pipelines (SDKv2, PF, dynamic) into a single provider surface. |
+| Dynamic bridge | `dynamic/*` | Parameterizable provider that downloads and hosts Terraform providers at runtime. |
+| Testing | `pkg/tests`, `pkg/internal/tests/cross-tests`, `pkg/pf/tests` | Harnesses that keep behavior aligned with Terraform and Pulumi expectations. |
+| Ops & Tooling | `Makefile`, `scripts/`, `docs/operations` | Build, lint, test, release, and operational playbooks. |
+
+## Build-Time Pipeline
+
+```
+Terraform Provider ──► Schema Introspection (`tfgen/schema.go`)
+      │                 │
+      │                 ├─► Pulumi PackageSpec (`pkg/tfgen/generate.go`)
+      │                 ├─► Language SDKs (`pkg/tfgen/generate_<lang>.go`)
+      │                 └─► Docs + Examples (`pkg/tfgen/docs.go`, `pkg/convert`)
+      ▼
+Pulumi SDK + Schema Artifacts
+```
+
+Key extension points:
+
+- **ProviderInfo overlays** (`pkg/tfbridge/info`) describe renames, doc overrides, examples, and custom CRUD behavior.
+- **Edit rules** (`pkg/tfgen/edit_rules.go`) post-process generated schema and docs.
+- **Converters** (`pkg/convert/*`) translate Terraform values to Pulumi `PropertyValue`s.
```

Comment:
pkg/convert is not important.. I would not highlight it here... I'd use package.go to document what the package is about.